### PR TITLE
Improve doc for torch.add and add doc for torch.sub.

### PR DESCRIPTION
**Pull request by xuhdev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25115 Add in-place operator rsub_.
* **#25114 Improve doc for torch.add and add doc for torch.sub.**

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25114

### CODE REVIEW

**Review by izdeby — APPROVED** [review]

**Review by gchanan — CHANGES_REQUESTED** [review]
Don't document this, it's considered deprecated, see https://github.com/pytorch/pytorch/blob/master/tools/autograd/deprecated.yaml.

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -210,6 +210,75 @@ def merge_dicts(*dicts):
...
+
+.. math::
+    \text{{out}} = \text{{input}} - \text{{other}}
+
+If :attr:`input` is of type FloatTensor or DoubleTensor, :attr:`other` must be
+a real number, otherwise it should be an integer.
+
+Args:
+    {input}
+    value (Number): the number to be added to each element of :attr:`input`
```
Comment (vishwakftw) [inline torch/_torch_docs.py]: `value` is not an argument for this variant I suppose.

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -210,6 +210,75 @@ def merge_dicts(*dicts):
             [ -8.9902,  -8.3667,  -7.3925,  -7.6147]])
 """.format(**common_args))
 
+add_docstr(torch.sub,
```
Comment (vishwakftw) [inline torch/_torch_docs.py]: It would also be nice if this could be placed in alphabetical order within the file.

**Comment by xuhdev** [comment]
> Don't document this, it's considered deprecated, see https://github.com/pytorch/pytorch/blob/master/tools/autograd/deprecated.yaml.

@gchanan I'm a bit confused. What is the replacement? How would one now, for example, do the fma operation (as `add` is also deprecated)?

File: torch/_torch_docs.py [hunk 3]
```diff
@@ -181,15 +155,15 @@ def merge_dicts(*dicts):
 .. math::
     \text{{out}} = \text{{input}} + \text{{alpha}} \times \text{{other}}
 
-If :attr:`other` is of type FloatTensor or DoubleTensor, :attr:`alpha` must be
-a real number, otherwise it should be an integer.
+If the result of adding :attr:`self` and :attr:`other` would not have a
```
Comment (gchanan) [inline torch/_torch_docs.py]: this isn't right after @nairbv's changes (also with sub below).  More accurate is something like "the category (floating point, integer, boolean)" of alpha must match that of the result of addition between `input` and `other`.

File: torch/_torch_docs.py [hunk 4]
```diff
@@ -181,15 +155,15 @@ def merge_dicts(*dicts):
 .. math::
     \text{{out}} = \text{{input}} + \text{{alpha}} \times \text{{other}}
 
-If :attr:`other` is of type FloatTensor or DoubleTensor, :attr:`alpha` must be
-a real number, otherwise it should be an integer.
+The category (floating point, integer, or Boolean) of :attr:`alpha` must match
+that of the result of addition between :attr:`input` and :attr:`other`.
```
Comment (zou3519) [inline torch/_torch_docs.py]: This isn't true. the following (integer alpha, float result) works for me on master:
```
import torch

x = torch.randn(3, 3)
y = torch.randn(3, 3)
torch.add(x, y, alpha=1)
```

Comment (zou3519) [reply torch/_torch_docs.py]: I would change this to say the following:

```
If the result of the addition between :attr:`input` and :attr:`other` is:
- floating point, :attr:`alpha` can be `float` or `int`
- integral, :attr:`alpha` must be `int`
- boolean, :attr:`alpha` can be `int` or `bool.
```

**Review by zou3519 — CHANGES_REQUESTED** [review]
"the category (floating point, integer, or Boolean) of :attr:`alpha` must match
a real number, otherwise it should be an integer.	that of the result of addition between :attr:`input` and :attr:`other`." is wrong, let's reword it.

File: torch/_torch_docs.py (hunk 4 above)
Comment (xuhdev) [reply torch/_torch_docs.py]: Thanks, I reworded a bit and updated this PR.

**Comment by xuhdev** [comment]
@gchanan I don't think so -- @nairv's PR is for the 1.3 branch, but this is for master

**Comment by xuhdev** [comment]
Of course, if you can cherry-pick the add and sub part of #27501 into master, I think this can be closed

**Comment by nairbv** [comment]
> Of course, if you can cherry-pick the add and sub part of #27501 into master, I think this can be closed

My PR on the 1.3 branch conflicts with master, so probably easier to land this one than to cherry pick that one.

**Comment by xuhdev** [comment]
Closed in favor of #28118, which also adds doc for `mul` and updates format starting from #27571

**Pull request closed without merging**